consensus/src/consensus/mod.rs

```diff
@@ -504,6 +509,52 @@ impl ConsensusApi for Consensus {
         self.headers_store.get_timestamp(self.get_sink()).unwrap()
     }
 
+    fn get_current_block_color(&self, hash: Hash) -> Option<bool> {
+        let mut queue: BinaryHeap<Reverse<SortableBlock>> = BinaryHeap::new();
+        let mut visited = BlockHashSet::new();
+
+        let initial_childrens = match self.get_block_children(hash) {
+            Some(children) => children,
+            None => return None,
+        };
+
+        for child in initial_childrens {
+            let block = self.get_block(child).unwrap();
```

Comment:
Ghostdag store has compact data access 

```suggestion
            let blue_work = self.ghostdag_primary_store.get_blue_work(child).unwrap();
```

The same applies below within the main loop